src/Steps/Release/BuildArchive.php

```diff
@@ -155,67 +125,31 @@ protected function write($path, $content)
      * Build a project of the given version in a temporary folder, and return the path to this
      *
      * @param OutputInterface $output
+     * @param Archive $archive
      * @return string Path to temporary project
+     * @throws Exception
      */
-    protected function createProject(OutputInterface $output)
+    protected function createArchiveFiles(OutputInterface $output, Archive $archive)
     {
-        // Get files
-        $version = $this->getVersion()->getValue();
-        $cmsArchive = "SilverStripe-cms-v{$version}";
-        $frameworkArchive = "SilverStripe-framework-v{$version}";
-
-        // Check path exists and is empty
-        $path = sys_get_temp_dir() . '/archiveTask';
-        $this->log($output, "Creating temporary project at {$path}");
-        $this->unlink($path);
-        mkdir($path);
+        $name = $archive->getRelease()->getLibrary()->getName();
+        $version = $archive->getRelease()->getVersion()->getValue();
+        $path = $archive->getTempDir();
+        $this->log($output, "Generating archives for <info>{$name}</info> at <comment>{$path}</comment>");
 
-        // Copy composer.phar
-        $this->log($output, "Getting composer.phar");
-        $this->copy('http://getcomposer.org/composer.phar', "{$path}/composer.phar");
+        // Ensure path is empty, but exists
+        $this->unlink($path);
+        if (!mkdir($path, 0777, true)) {
```

Comment:
does it need to be `0777`? would `0775` or `0755` be sufficient for this temp folder?

Reply:
Fixed.